man/man1/pmlogger_daily_report.1
'\"macro stdmacro
.\"
.\" Copyright (c) 2018-2019 Red Hat.
.\"
.\" This program is free software; you can redistribute it and/or modify it
.\" under the terms of the GNU General Public License as published by the
.\" Free Software Foundation; either version 2 of the License, or (at your
.\" option) any later version.
.\"
.\" This program is distributed in the hope that it will be useful, but
.\" WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
.\" or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
.\" for more details.
.\"
.TH PMLOGGER_DAILY_REPORT 1 "PCP" "Performance Co-Pilot"
.SH NAME
\f3pmlogger_daily_report\f1 \- write Performance Co-Pilot daily summary reports
.SH SYNOPSIS
.B $PCP_BINADM_DIR/pmlogger_daily_report
[\f3\-ApV?\f1]
[\f3\-a\f1 \f2archivefile\f1]
[\f3\-f\f1 \f2outputfile\f1]
[\f3\-h\f1 \f2hostname\f1]
[\f3\-l\f1 \f2logfile\f1]
[\f3\-o\f1 \f2directory\f1]
[\f3\-t\f1 \f2interval\f1]
.SH DESCRIPTION
.B pmlogger_daily_report
writes daily performance summary reports, much like those produced by
.BR sadc (1)
and the
.BR sa2 (8)
utility.
.PP
All of the command line arguments are optional and intended to be self
explanatory.
.PP
By default, the reports are not generated, but
if the
.I pcp-zeroconf
package has been installed (there will be files
in
.BR $PCP_VAR_DIR/config/pmlogconf/zeroconf )
then the necessary metrics will have been included in the default
.BR pmlogger (1)
configuration file and the reports will be generated.
.PP
The reports are generated (by default) soon after midnight, once
.BR pmlogger_daily (1)
has completed the merging of any partial archives to generate one
archive for yesterday's activity.
.PP
Each performance summary report is named
.BI sar XX
(where
.I XX
is yesterdays day-of-the-month).
The
.I outputfile
may be changed with the
.B \-f
option.
The report will be written to the
.I $PCP_LOG_DIR/sa
directory by default, but this may be changed with the
.B \-o
option to a different
.IR directory .
.PP
Note that there are sufficiently flexible command line options for
.B pmlogger_daily_report
to be used to read any
.I archivefile
and write the report to any output directory.
.PP
If the
.B \-a
option is not given, the default input
.I archivefile
is
.IR $PCP_ARCHIVE_DIR/HOSTNAME/YYYYMMDD ,
where
.I HOSTNAME
defaults to the local hostname (unless changed with the
.B \-h
option) and
.I YYYYMMDD
is the base name of yesterdays merged archive, as produced by
.BR pmlogger (1)
and
.BR pmlogger_daily (1).
If
.I archivefile
is a directory, then
.B pmlogger_daily_report
will use all PCP archives found in that directory
to write the report (this is known as multi-archive mode,
and may be considerably slower than specifying a single
archive as the input).
.PP
The reports themselves are created by the
.BR pmrep (1)
utility using its default configuration file, see
.BR pmrep.conf (5).
The
.BR pmrep (1)
configuration entries used to write the reports is currently hardwired
into the
.B pmlogger_daily_report
script.
.PP
Finally, the input archives must contain sufficient metrics as needed by
.BR pmrep (1)
to write the report.
On platforms that support it, the
.I pcp-zeroconf
package configures PCP logging as required for this \- hence
.B pmlogger_daily_report
should be used with the
.BR pmlogger (1)
configuration that is set up by
.BR pcp-zeroconf .
As the name suggests,
.BR pcp-zeroconf
requires no additional configuration after installation in order to
capture the required archives needed by
.BR pmlogger_daily_report .
.SH OPTIONS
The available command line options are:
.TP 5
\fB\-a\fR \fIarchive\fR
Specifies an alternate input
.I archive
file basename or directory path.
.TP
\fB\-A\fR
Use the start and end times of input archive for the report, as
opposed to the default behaviour of 24 hours from midnight yesterday.
.TP
\fB\-f\fR \fIfilename\fR
Specifies an alternate output
.IR filename .
\fB\-h\fR \fIhostname\fR
Specifies an alternate
.I hostname
to use within the default input archive file path.
.TP
\fB\-l\fR \fIfile\fR, \fB\-\-logfile\fR=\fIfile\fR
In order to ensure that mail is not unintentionally sent when this
script is run from
.BR pmlogger_daily (1)
diagnostics are always sent to a log
.IR file .
By default, this file is
.I $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.log
but this can be changed using the
.B \-l
option.
If this log file already exists when the script starts, it will be
renamed with a
.I .prev
suffix (overwriting any log file saved earlier) before diagnostics
are generated to the log file.
.TP
\fB\-p\fR
If this option is specified
then the status of the daily processing is polled and if the
report has not been done in the last 24 hours then it is done now.
With this option,
.B pmlogger_daily_report
simply exits if the previous day's processing has already been
done.
.RS
.PP
This option is no longer required as the polling of
.BR pmlogger_daily (1)
will ensure the report will be generated, however the option is retained
for backwards compatibility.
.RE
.TP
\fB\-t\fR \fIinterval\fR
Specifies the sampling
.I interval
used when generating the report, in the format described in
.BR PCPIntro (1).
The default is every 10 minutes.
.TP
\fB\-V\fR, \fB\-\-verbose\fR
The output from the execution of the script may be extended using
this option which enables verbose tracing of activity.
By default the script generates no log output unless some error or
warning condition is encountered.
.TP
\fB\-?\fR, \fB\-\-help\fR
Display usage message and exit.
.SH PCP ENVIRONMENT
Environment variables with the prefix \fBPCP_\fP are used to parameterize
the file and directory names used by PCP.
On each installation, the
file \fI/etc/pcp.conf\fP contains the local values for these variables.
The \fB$PCP_CONF\fP variable may be used to specify an alternative
configuration file, as described in \fBpcp.conf\fP(5).
.SH SEE ALSO
.BR PCPIntro (1),
.BR pmlogger_daily (1),
.BR pmlogger (1),
.BR pmrep (1),
.BR sadc (1)
and
.BR sa2 (8).

.\" control lines for scripts/man-spell
.\" +ok+ YYYYMMDD prev {from .prev suffix} sa sadc zeroconf


CHANGELOG
pcp-6.3.6 (17 March 2025)
    - Debian build changes - see debian/changelog for details

pcp-6.3.4 (13 March 2025)
    - libpcp: fix endian issue in big-endian __pmTimestamp writing
    - pcp-xsos: fine-tune error handling on bad command line options
    - pcp-xsos: single-line pmPrintValue/pminfo values, escape shell chars
    - pcp-atop: correct BPF per-process network metric names
    - pmcd, pmdaproc: allow non-local username connection attributes
    - pmlogger, libpcp: fix handling of dynamic instance domain labels
    - pmrep: use BPF instead of BCC for proc networking metrics
    - pmlogconf: use libbpf over BCC for networking metrics
    - pmlogextract: fix logic around instances labels, allowing set updates
    - pmdalinux: mem.numa.hugepages metrics (per node, per hugepage size)
    - pmdabpftrace: fix for some python3 versions async IO create_task
    - pmdanvidia: if present use dso from the libnvidia-ml runtime package
    - pmdadm, pmieconf: default to enabling the dmthin pmie rules
    - configure.ac: fix autoreconf, add AC_DEFINE_UNQUOTED descriptions
    - rpm builds: enable bpf, bpftrace, bcc on riscv64 builds
    - rpm builds: drop user creation scriptlets for rpm builds
    - deb build: force debian/rules to be recreated each time
    - containers: updates to the archive-analysis container
    - selinux: several policy updates for new AVC denials
    - docs: improve SSL/TLS diagnostic in pcp-verify script

pcp-6.3.3 (29 January 2025)
    - Debian build changes - see debian/changelog for details

pcp-6.3.3 (29 January 2025)
    - Debian build changes - see debian/changelog for details

pcp-6.3.2 (6 November 2024)
    - PMDA additions, enhancements and bug fixes:
	pmdabpf: avoid a too-low rlimit setting within pmdabpf
	pmdaproc: add an access control mechanism for remote sampling
	pmdalinux: ensure divide-by-zero cannot happen with tmpfs.full
	pmdalinux: additional MPTcpExt netstat metrics
	pmdastatsd: keep correct track of pmns mapped metric count
	pmdaopenmetrics: run as pcp user rather than root by default
	pmdaopenmetrics: ensure correct cache file permissions used
    - Server-side utilities and log management scripts:
	libpcp, pmlogger: add support for zstd(1) compression of archives
	pmlogbasename: new, maps filename to PCP archive basename
	pmlogcompress: new, archive compression/decompression utility
	pmlogcheck: performance improvements
	pmlogcheck:  allow multiple archive options on command line
	pmlogextract: extend -v option with the full pmlogger syntax
	pmlogextract: fix for reporting file size on 32-bit platforms
    - libpcp, libpcp_pmda, libpcp_mmv, libpcp_web and language bindings:
	libpcp: fix 64-bit big endian v3 archive timestamp decoding
	libpcp: change in archive value interpolation for non-counter metrics
	python api: allow timeval and timespec init with float/datetime
    - Client tools and utilities:
	pminfo: minor error path reworking
	pcp-buddyinfo: addition of a --no-interpol option
	pcp-ps: added option to view full task name with its argument
    - Build, infrastructure and packaging updates:
	build: deal with botched deb package names from Debian
	build: ensure pmie config.default created with correct ownership
	systemd: move pcp-reboot-init rpm spec logic from zeroconf to pcp
	selinux: ensure farm pmlogger/pmie processes labelled correctly
    - Documentation:
	docs: refine pmdaproc -A description details
	docs: make pmstat(1) match the code for -t defaults

pcp-6.3.1 (17 September 2024)
    - Client tools and utilities:
	pcp2arrow: fix extraction of values for metrics with indoms
	pcp-htop: fix platform setting enabling screen tabs
	pcp-htop: update to latest upstream bug fixes
	pcp-meminfo: added timestamp string for consistency
    - PMDA additions, enhancements and bug fixes:
	pmdaopenmetrics: add vLLM metrics support by default
	pmdahacluster: handle pacemaker 2.1.6 crm_mon output changes
	pmdahacluster: handle new style Pacemaker role descriptions
	pmdalinux: MPJoinSynBackupRx, MPJoinSynAckBackupRx netstat metrics
	pmdalinux: sysfs fallback from hinv.cpu.clock metric
	pmdaproc: lazy initialisation for accounting metrics
	pmdapmcd: expose pmcd limits as new metrics
	pmdammv: use isnan instead of open-coding NaN checks
    - Server-side utilities and log management scripts:
	pmcd: put an explicit cap on max number of fetched PMIDs
	pmcd: put an explicit cap on max number of contexts per client
	pmcd: cleanup caching of local hostname string
	pmpost: guard against possible symlink attack
	pmproxy: transition to dual Redis/Valkey support
    - libpcp, libpcp_pmda, libpcp_mmv, libpcp_web and language bindings:
	libpcp_web: fix a race in webgroup context teardown on timeout
	libpcp_web: transition to dual Redis/Valkey support
	libpcp: auto-load the proc PMDA in local context mode
	libpcp: fix buffer overflow in __pmServerAddInterface
	libpcp: refactor the attr PDU content checking
	libpcp: desc PDU consistency check and error message rework
	libpcp: text PDU consistency check and error message rework
	libpcp: creds PDU consistency check and error message rework
	libpcp: label PDU consistency check and error message rework
	libpcp: instance PDU consistency check and error message rework
	libpcp: rework some fetch PDU integrity checks and error messages
	libpcp: reworked __pmDecodeIDList integrity checks and warnings
	libpcp: ensure no fd leakage to child from __pmProcessPipe
	libpcp: ensure container names contain limited characters
	libpcp: fix buffer overrun test in __pmDecodeValueSet
	libpcp: hardening of the results PDU handling
	libpcp: hardening of the creds PDU handling
	libpcp: add new __pmCtlDebug() function
	python API: fix MMV metric text, indom and semantics handling
	python API: add log import timestamp convenience mechanisms
    - Build, infrastructure and packaging updates:
	build: tweak Makepkgs as dpkg-parsechangelog(1) is a bit picky
	build: drop gfs2 PMDA from build if distro vendor opts-out

src/pmlogger/pmlogger_daily_report.sh
#! /bin/sh
#
# Copyright (c) 2018 Red Hat.
#
# This program is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by the
# Free Software Foundation; either version 2 of the License, or (at your
# option) any later version.
#
# This program is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
# for more details.
#
#
# Administrative script for daily sar-style report summaries.
# This is intened to be run from cron, an hour or two after the pmlogger_daily
# cron script has completed, e.g. around 2am would be a suitable time.
#
# Sample crontab entry:
#
# # daily generation of system activity reports
# 0     2  *  *  *  pcp  /usr/libexec/pcp/bin/pmlogger_daily_report
#
#
# By default, the daily report will be written to /var/log/pcp/sa/sarXX
# where XX is the day of the month, yesterday.
#

. $PCP_DIR/etc/pcp.env
. $PCP_SHARE_DIR/lib/utilproc.sh

status=0
prog=`basename $0`

# optional begin logging to $PCP_LOG_DIR/NOTICES
#
if $PCP_LOG_RC_SCRIPTS
then
    logmsg="begin pid:$$ $prog args:$*"
    if which pstree >/dev/null 2>&1
    then
	logmsg="$logmsg [`_pstree_oneline $$`]"
    fi
    $PCP_BINADM_DIR/pmpost "$logmsg"
fi

# error messages should go to stderr, not the GUI notifiers
unset PCP_STDERR

# want mode for mkdir below to reliably be rwxrwxr-x (775)
#
umask 022

# default message log file
PROGLOG=$PCP_LOG_DIR/pmlogger/$prog.log
USE_SYSLOG=true
tmp=`mktemp -d "$PCP_TMPFILE_DIR/pmlogger_daily_report.XXXXXXXXX"` || exit 1

_cleanup()
{
    $USE_SYSLOG && [ $status -ne 0 ] && \
    $PCP_SYSLOG_PROG -p daemon.error "$prog failed - see $PROGLOG"
    if [ "$PROGLOG" != "/dev/tty" ]
    then
	[ -s "$PROGLOG" ] || rm -f "$PROGLOG"
    fi
    rm -rf $tmp
}
trap "_cleanup; exit \$status" 0 1 2 3 15

echo > $tmp/usage
cat >> $tmp/usage <<EOF
Options:
  -a=FILE		  input archive file basename or directory path (default yesterdays $PCP_ARCHIVE_DIR/<hostname>/YYYYMMDD)
  -f=FILE		  output filename (default "sarXX" in directory specified with -o, for XX yesterdays day of the month)
  -h=HOSTNAME		  hostname, affects the default input archive file path (default local hostname)
  -l=FILE,--logfile=FILE  send important diagnostic messages to FILE
  -p                      poll and exit if processing already done for today
  -o=DIRECTORY		  output directory (default $PCP_SA_DIR, see also -f option)
  -t=TIME		  reporting interval, default 10m
  -A			  use start and end times of input archive for the report (default midnight yesterday for 24hours)
  -V,--verbose            verbose messages in log (multiple times for very verbose), see also -l option
  --help
EOF

_usage()
{
    pmgetopt --progname=$prog --config=$tmp/usage --usage
    status=1
    exit
}

# option parsing
#
VERBOSE=false
ARCHIVETIMES=false
VERY_VERBOSE=false
PFLAG=false
MYARGS=""
DEBUG=false		# desperate debugging

ARGS=`pmgetopt --progname=$prog --config=$tmp/usage -- "$@"`
[ $? != 0 ] && exit 1

eval set -- "$ARGS"
while [ $# -gt 0 ]
do
    case "$1"
    in
	-a)	ARCHIVEPATH="$2"
		$DEBUG && echo "debug: after -a:  ARCHIVEPATH=$ARCHIVEPATH"
		shift
		;;
	-f)	REPORTFILE="$2"
		shift
		;;
	-h)	HOSTNAME="$2"
		shift
		;;
	-l)	PROGLOG="$2"
		USE_SYSLOG=false
		shift
		;;
	-o)	REPORTDIR="$2"
		shift
		;;
	-p)     PFLAG=true
		;;
	-t)	INTERVAL="$2"
		shift
		;;
	-A)	ARCHIVETIMES=true
		;;
	-V)	if $VERBOSE
		then
		    VERY_VERBOSE=true
		else
		    VERBOSE=true
		fi
		MYARGS="$MYARGS -V"
		;;
	-\?)	_usage
		;;
    esac
    shift
done

[ $# -ne 0 ] && _usage

# optionally uncompress a compressed archive
#
_uncompress()
{
    if pmlogdecompress -t $tmp "$1"
    then
	echo $tmp/`basename "$1"`
    else
	$VERBOSE && echo >&2 "Uncompressing failed, reverting to compressed archive"
	echo "$1"
    fi
}

if $PFLAG
then
    rm -f $tmp/ok
    if [ -f $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp ]
    then
	last_stamp=`sed -e '/^#/d' <$PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp`
	if [ -n "$last_stamp" ]
	then
	    # Polling happens every 60 mins, so if pmlogger_daily was last
	    # run more than 23.5 hours ago, we need to do it again, otherwise
	    # exit quietly
	    #
	    now_stamp=`pmdate %s`
	    check=`expr $now_stamp - \( 23 \* 3600 \) - 1800`
	    if [ "$last_stamp" -ge "$check" ]
	    then
		# nothing to be done, yet
		exit
	    fi
	    touch $tmp/ok
	fi
    fi
    if [ ! -f $tmp/ok ]
    then
	# special start up logic when pmlogger_daily_report.stamp does not
	# exist ... punt on archive files being below $PCP_LOG_DIR/sa
	#
	if [ -d $PCP_LOG_DIR/sa ]
	then
	    find $PCP_LOG_DIR/sa -name "sar`pmdate -1d %d`" >$tmp/tmp
	    if [ -s $tmp/tmp ]
	    then
		# heuristic match, assume nothing to be done
		exit
	    fi
	fi
    fi
fi

# if it looks like we can create/update the date-and-timestamp
# then do it ... this will be checked by -p polling
#
if [ -w $PCP_LOG_DIR/pmlogger ]
then
    if [ ! -f $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp \
           -o -w $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp ]
    then
	if _save_prev_file $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp
	then
	    :
	else
	    echo "Warning: cannot save previous date-and-timestamp" >&2
	fi
	# only update date-and-timestamp if we can write the file
	#
	pmdate '# %Y-%m-%d %H:%M:%S
%s' >$tmp/stamp
	if cp $tmp/stamp $PCP_LOG_DIR/pmlogger/pmlogger_daily_report.stamp
	then
	    :
	else
	    echo "Warning: cannot install new date-and-timestamp" >&2
	fi
    fi
fi

if [ "$PROGLOG" = "/dev/tty" ]
then
    # special case for debugging ... no salt away previous
    #
    :
else
    # After argument checking, everything must be logged to ensure no mail is
    # accidentally sent from cron.  Close stdout and stderr, then open stdout
    # as our logfile and redirect stderr there too.
    #
    [ -f "$PROGLOG" ] && mv "$PROGLOG" "$PROGLOG.prev"
    exec 1>"$PROGLOG" 2>&1
fi

# Default hostname is the name of the local host
#
[ -z "$HOSTNAME" ] && HOSTNAME=`hostname`

# Default input archive is the merged archive for yesterday.
# Unfortunately we can't just specify the entire directory
# and rely on multi-archive mode because this script might
# take a long time to run as a result.
#
[ -z "$ARCHIVEPATH" ] && ARCHIVEPATH=$PCP_ARCHIVE_DIR/$HOSTNAME/`pmdate -1d %Y%m%d`
$DEBUG && echo "debug: after default:  ARCHIVEPATH=$ARCHIVEPATH"

# If input archive has been compressed, then uncompress it
# into temporary files and use these, to avoid repeated
# uncompressing for pmrep below.
#
ORIG_ARCHIVEPATH="$ARCHIVEPATH"
ARCHIVEPATH=`_uncompress $ARCHIVEPATH`
$DEBUG && echo "debug: after _uncompress:  ARCHIVEPATH=$ARCHIVEPATH"
$VERBOSE && echo ARCHIVEPATH=$ARCHIVEPATH

# Default output directory
#
[ -z "$REPORTDIR" ] && REPORTDIR="$PCP_SA_DIR"
$VERBOSE && echo REPORTDIR=$REPORTDIR

# Create output directory - if this fails for any reason we exit later
#
# mode rwxrwxr-x is the default for pcp:pcp dirs
umask 002
[ -d "$REPORTDIR" ] \
    || mkdir -p -m 0775 "$REPORTDIR" 2>/dev/null
# reset the default mode to rw-rw-r- for files
umask 022

# Default output file is the day of month for yesterday in REPORTDIR
#
[ -z "$REPORTFILE" ] && REPORTFILE=$REPORTDIR/sar`pmdate -1d %d`
$VERBOSE && echo REPORTFILE=$REPORTFILE

# Default reporting interval is 10m (same as sysstat uses)
#
[ -z "$INTERVAL" ] && INTERVAL="10m"

# If the input archive doesn't exist, we exit.
#
if [ ! -f "`echo $ARCHIVEPATH.meta*`" ]; then
    # report this to the log
    $DEBUG && ls -l $ARCHIVEPATH.*
    echo "$prog: FATAL error: Failed to find input archive \"$ARCHIVEPATH\""
    exit 1
fi

# Common reporting options, including time window: midnight yesterday for 24h
#
REPORT_OPTIONS="-a $ARCHIVEPATH -z -E 0 -p -f%H:%M:%S -t$INTERVAL"
if ! $ARCHIVETIMES
then
    # specific start/finish times - midnight yesterday for 24h
    start=`pmdate -1d "@%d-%b-%Y"`
    REPORT_OPTIONS="$REPORT_OPTIONS -S$start -T 24h"
fi
$VERBOSE && echo REPORT_OPTIONS=$REPORT_OPTIONS

# Truncate or create the output file. Exit if this fails.
#
if ! cp /dev/null $REPORTFILE
then
    status=$?
    echo "$prog: FATAL error: cannot create \"$REPORTFILE\""
    exit
fi

#
# Common reporting function
#
_report()
{
    _conf=$1
    _comment="$2"

    $VERBOSE && echo Generating report for $_conf $_comment
    echo >>$REPORTFILE; echo >>$REPORTFILE
    pmdumplog -z -l $ARCHIVEPATH | awk '/commencing/ {print "# ",$2,$3,$4,$5,$6}' >>$REPORTFILE
    echo $_comment >>$REPORTFILE
    $VERBOSE && echo pmrep $REPORT_OPTIONS $_conf
    pmrep $REPORT_OPTIONS $_conf >$tmp/out 2>$tmp/err
    if [ -s $tmp/out ]
    then
    	cat $tmp/out >>$REPORTFILE
    else
	if grep 'PM_ERR_NAME' $tmp/err >/dev/null 2>&1
	then
	    metric=`$PCP_AWK_PROG <$tmp/err '/PM_ERR_NAME/ { print $3; exit }'`
	    echo "-- no report for config \"$_conf\" because the metric \"$metric\" is not in the archive" >>$REPORTFILE
	elif grep 'PM_ERR_INDOM_LOG' $tmp/err >/dev/null 2>&1
	then
	    metric=`$PCP_AWK_PROG <$tmp/err '/PM_ERR_INDOM_LOG/ { print $3; exit }'`
	    echo "-- no report for config \"$_conf\" because there are no values for any instance of the metric \"$metric\" in the archive" >>$REPORTFILE
	elif grep 'PM_ERR_BADDERIVE' $tmp/err >/dev/null 2>&1
	then
	    metric=`$PCP_AWK_PROG <$tmp/err '/PM_ERR_BADDERIVE/ { print $3; exit }'`
	    echo "-- no report for config \"$_conf\" because one or more metrics for the derived metric \"$metric\" is not in the archive" >>$REPORTFILE
	else
	    cat $tmp/err >>$REPORTFILE
	    echo "-- no report for config \"$_conf\"" >>$REPORTFILE
	fi
    fi
}

echo "System Activity Report" >>$REPORTFILE
echo >>$REPORTFILE
echo "Host:            $HOSTNAME" >>$REPORTFILE
echo "Archive:         $ORIG_ARCHIVEPATH" >>$REPORTFILE
echo "Report created:  `date`" >>$REPORTFILE

_report :sar-u-ALL '# CPU Utilization statistics, all CPUS'
_report :sar-u-ALL-P-ALL '# CPU Utilization statistics, per-CPU'
_report :vmstat '# virtual memory (vmstat) statistics'
_report :vmstat-a '# virtual memory active/inactive memory statistics'
_report :sar-B '# paging statistics'
_report :sar-b '# I/O and transfer rate statistics'
_report :sar-d-dev '# block device statistics'
_report :sar-d-dm '# device-mapper device statistics'
_report :sar-F '# mounted filesystem statistics'
_report :sar-H '# hugepages utilization statistics'
_report :sar-I-SUM '# interrupt statistics, summed'
_report :sar-n-DEV '# network statistics, per device'
_report :sar-n-EDEV '# network error statistics, per device'
_report :sar-n-NFSv4 '# NFSv4 client and RPC statistics'
_report :sar-n-NFSDv4 '# NFSv4 server and RPC statistics'
_report :sar-n-SOCK '# socket statistics'
_report :sar-n-TCP-ETCP '# TCP statistics'
_report :sar-q '# queue length and load averages'
_report :sar-r '# memory utilization statistics'
_report :sar-S '# swap usage statistics'
_report :sar-W '# swapping statistics'
_report :sar-w '# task creation and system switching statistics'
_report :sar-y '# TTY devices activity'
_report :numa-hint-faults '# NUMA hint fault statistics'
_report :numa-per-node-cpu '# NUMA per-node CPU statistics'
_report :numa-pgmigrate-per-node '# NUMA per-node page migration statistics'

# optional end logging to $PCP_LOG_DIR/NOTICES
#
if $PCP_LOG_RC_SCRIPTS
then
    $PCP_BINADM_DIR/pmpost "end pid:$$ $prog status=$status"
fi

exit
